projects/step-frontend/src/lib/modules/timeseries/performance-view/filter-bar/filter-bar.component.ts

```diff
@@ -46,19 +48,68 @@ export class FilterBarComponent implements OnInit, OnDestroy {
   filters: TsFilterItem[] = [];
   moreFilters: TsFilterItem[] = [];
 
-  constructor(private _changeDetectorRef: ChangeDetectorRef) {}
+  rawMeasurementsModeActive = false;
+  showApplyButton = false;
+
+  constructor(private _changeDetectorRef: ChangeDetectorRef, private timeSeriesService: TimeSeriesService) {}
 
   ngOnInit(): void {
     if (!this.context) {
       throw new Error('Context input is mandatory');
     }
     this.filters = this.activeFilters;
     this.emitFilterChange$.pipe(debounceTime(this.EMIT_DEBOUNCE_TIME)).subscribe(() => {
-      // we ignore empty filters
-      this.onFiltersChange.emit(this.filters.filter(FilterUtils.filterItemIsValid));
+      const filters = this.filters.filter(FilterUtils.filterItemIsValid);
+      if (filters.length) {
+        this.showApplyButton = true;
+        this.composeAndVerifyOql(filters).subscribe((response) => {
+          this.rawMeasurementsModeActive = response.hasUnknownFields;
+          this.showApplyButton = response.hasUnknownFields;
```

Comment:
What is the reasoning behind showing the button while the request is ongoing?

Reply:
you are right. it's better to have a bit of delay instead